$MirOS: ports/emulators/qemu/patches/patch-block-dmg_c,v 1.1 2005/05/03 19:30:03 tg Exp $
--- block-dmg.c.orig	Sun Sep  4 17:11:09 2005
+++ block-dmg.c	Tue Nov 15 11:37:42 2005
@@ -228,9 +228,9 @@ static inline int dmg_read_chunk(BDRVDMG
 	    if (ret != s->lengths[chunk])
 		return -1;
 	
-	    s->zstream.next_in = s->compressed_chunk;
+	    s->zstream.next_in = (unsigned char *)s->compressed_chunk;
 	    s->zstream.avail_in = s->lengths[chunk];
-	    s->zstream.next_out = s->uncompressed_chunk;
+	    s->zstream.next_out = (unsigned char *)s->uncompressed_chunk;
 	    s->zstream.avail_out = 512*s->sectorcounts[chunk];
 	    ret = inflateReset(&s->zstream);
 	    if(ret != Z_OK)
@@ -293,5 +293,9 @@ BlockDriver bdrv_dmg = {
     dmg_read,
     NULL,
     dmg_close,
+    NULL,
+    NULL,
+    NULL,
+    NULL
 };
 
